src/kibana/components/agg_types/buckets/terms.js

```diff
@@ -146,11 +137,16 @@ define(function (require) {
               output.params.valueType = agg.field().type === 'number' ? 'float' : agg.field().type;
             }
 
-            if (!orderAgg || orderAgg.type.name === 'count') {
+            if (orderAgg && orderAgg.type.name === 'count') {
               order._count = dir;
               return;
             }
 
+            if (!orderAgg) {
```

Comment:
Can you move this up so it returns first when met, and remove the `orderAgg &&` bit from what is currently line 140?